In [1]:
import MySQLdb
import pandas as pd

In [2]:

cloud_drugs = {}
fp = open('../data/CLOUD_drugs.txt')
for line in fp:
    tmp = line.strip().split('\t')
    cloud_drugs[tmp[1]] = tmp[0]

print len(cloud_drugs)

267


In [17]:
db = MySQLdb.connect("menchelabdb.int.cemm.at","root","cqsr4h","ImageAnalysisDDI" )

In [18]:
CLOPAC_to_CLOUD_IDS = {}
CLOPAC_CLOUD_IDS = []
for cloud_name in cloud_drugs.keys():
    #print cloud_name
    string = 'select Image_Metadata_ID_A,Image_Metadata_Name_A from IsabelCLOUPAC_Per_Image where Image_Metadata_Name_A like "'+cloud_name+'" group by (Image_Metadata_ID_A);'
    
    found_CLOUPAC = pd.read_sql(string, con=db)
    
    if len(found_CLOUPAC) == 1:
    
        CLOPAC_CLOUD_IDS.append(found_CLOUPAC['Image_Metadata_ID_A'].values[0])
        CLOPAC_to_CLOUD_IDS[cloud_drugs[cloud_name]] = found_CLOUPAC['Image_Metadata_ID_A'].values[0]
    elif len(found_CLOUPAC) > 1:

        valid_ids = found_CLOUPAC['Image_Metadata_ID_A'].values
        valid_ids = [x for x in valid_ids if 'CLOUD' in x]
        
        if len(valid_ids) ==1:
            CLOPAC_to_CLOUD_IDS[cloud_drugs[cloud_name]] = valid_ids[0]
            CLOPAC_CLOUD_IDS.append(valid_ids[0])
        else:
            print '-------'
            print 'Problem with: ' + cloud_name 
            print '-------'
    else:
        print '-------'
        print 'Problem with: ' + cloud_name 
        print '-------'

dp.close()
print CLOPAC_CLOUD_IDS

-------
Problem with: Pyrazinamide
-------
-------
Problem with: Atenolol
-------
-------
Problem with: Thalidomide
-------
-------
Problem with: Teriflunomide (A77 1726)
-------
-------
Problem with: Azacitidine
-------
-------
Problem with: Baclofen
-------
-------
Problem with: Zalcitabine
-------
-------
Problem with: Triflupromazine
-------
-------
Problem with: Trifluoperazine
-------
-------
Problem with: Disulfiram
-------
-------
Problem with: Naltrexone
-------
-------
Problem with: Methyldopa
-------


NameError: name 'dp' is not defined

Pyrazinamide = P7136 (Pyrazinecarboxamide)  
Atenolol = A7655, A143 ((+/-)-Atenolol and S(-)-Atenolol)  
Thalidomide = T144 ((+/-)-Thalidomide)  
Teriflunomide (A77 1726) = SMl0936 (Teriflunomide)  
Azacitidine = A2385 (5-azacytidine)  
Lumefantrine = L5420, CLOUD130 (Lumefantrine)  
Baclofen = B5399 ((+/-)-Baclofen)  
Zalcitabine = D5782 (2';3'-dideoxycytidine)  
Triflupromazine = T2896 (Triflupromazine)  
Succinylcholine Chloride = S8251, CLOUD261 (Succinylcholine chloride)  
Trifluoperazine = T8516 (Trifluoperazine dihydrochloride)  
Disulfiram = T1132  (Tetraethylthiuram disulfide)
Naltrexone = N3136 (Naltrexone hydrochloride)  
Carbachol = C4382, CLOUD012 (Carbachol)  
Methyldopa = D1507 (L-3;4-Dihydroxyphenylalanine methyl ester hydrochloride)  

In [19]:
print len(CLOPAC_CLOUD_IDS)

255


In [20]:
# all possibilities
#CLOPAC_CLOUD_IDS.extend(['P7136','A7655','A143','T144','SMl0936','A2385','L5420','CLOUD130','B5399','D5782','T2896','S8251','CLOUD261','T8516','N3136','C4382','CLOUD012','D1507'])

CLOPAC_CLOUD_IDS.extend(['P7136','A7655','T144','SMl0936','A2385','B5399','D5782','T2896','T8516','T1132','N3136','D1507'])


In [21]:
print len(CLOPAC_CLOUD_IDS)

267


In [22]:
fp_out = open('../results/IsabelCLOUPAC_Per_Image/List_of_CLOUD_Drugs/List_of_usable_IDs_267.csv','w')
for CLOUD_ID in CLOPAC_CLOUD_IDS:
    fp_out.write(CLOUD_ID+'\n')
fp_out.close()